dbms/src/Flash/Coprocessor/RemoteRequest.cpp

```diff
@@ -36,13 +42,14 @@ RemoteRequest RemoteRequest::build(const RegionRetryList & retry_regions, DAGCon
     DAGSchema schema;
     tipb::DAGRequest dag_req;
     auto * executor = dag_req.mutable_root_executor();
-    if (selection != nullptr)
+
+    if (push_down_filter.hasValue())
```

Comment:
nit: How about extract the logic into PushDownFilter，the function name can be PushDownFilter::buildSelection

Reply:
ok